sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/ReferenceManagerImpl.java

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.implementation;
+
+import com.azure.core.util.ReferenceManager;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Implementation of {@link ReferenceManager}.
+ */
+public final class ReferenceManagerImpl implements ReferenceManager, Runnable {
+    // Used to create a unique thread which is used to clean up the resources managed by an instance ReferenceManager.
+    private static final AtomicInteger RESOURCE_MANAGER_THREAD_NUMBER = new AtomicInteger();
+
+    // Base name for ResourceManager threads.
+    private static final String BASE_THREAD_NAME = "azure-sdk-referencemanager-";
+
+    private final CleanableReference<?> cleanableReferenceList;
+    private final ReferenceQueue<Object> queue;
+
+    /**
+     * Creates a new instance of {@link ReferenceManagerImpl}.
+     */
+    public ReferenceManagerImpl() {
+        this.queue = new ReferenceQueue<>();
+        this.cleanableReferenceList = new CleanableReference<>();
+
+        new CleanableReference<>(this, () -> { }, this);
+
+        Thread thread = new Thread(Thread.currentThread().getThreadGroup(), this,
+            BASE_THREAD_NAME + RESOURCE_MANAGER_THREAD_NUMBER.getAndIncrement());
```

Comment:
In what context is the `ReferenceManagerImpl` constructor called more than once? It appears to be a static field on `ReferenceManager`, and so I'm unsure why we need the thread number incrementing?

Reply:
Hold over from when multiple instance could be created, I'll comment this out and simplify it